scalameta/src/main/scala/scala/meta/Trees.scala

```diff
@@ -195,7 +200,7 @@ package scala.meta.internal.ast {
       // require(keywords.contains(value) ==> isBackquoted)
     }
     @ast class Select(qual: Term.Ref, name: Type.Name) extends Type.Ref with Pat.Type.Ref {
-      require(qual.isPath || qual.isInstanceOf[Term.Super])
+      require(qual.isPath || qual.isInstanceOf[Term.Super] || qual.isInstanceOf[Term.Quasi])
```

Comment:
Why do we allow `qual` to be `Term.Quasi`? Why not make it `become` a `Term.Ref.Quasi`?